plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -609,6 +607,29 @@ impl FirmwareManager {
                             && recorded_entry.version == smartrest_request.version
                             && recorded_entry.server_url == smartrest_request.url
                         {
+                            info!("The same operation as the received c8y_Firmware operation is already in progress.");
+
+                            // Resend a firmware request with incremented attempt.
+                            let new_operation_entry = recorded_entry.increment_attempt();
+                            new_operation_entry.overwrite_file(persistent_store.clone())?;
+
+                            let request =
+                                FirmwareOperationRequest::from(new_operation_entry.clone());
+                            self.mqtt_client.published.send(request.try_into()?).await?;
+                            info!(
+                                "Firmware update request is resent. operation_id={}, child={}",
+                                new_operation_entry.operation_id, new_operation_entry.child_id
+                            );
+
+                            self.operation_timer.start_timer(
+                                (
+                                    new_operation_entry.child_id,
+                                    new_operation_entry.operation_id,
+                                ),
+                                ActiveOperationState::Pending,
```

Comment:
It may not be a good idea to forcefully replace the existing `ActiveOperationState` with `Pending`.  What if the child had already moved it to `Executing`?

Reply:
Definitely you are right. I'll fix.